File: golang/Lab_4.go
package Lab_4

import (
	"fmt"
	"math"
)

func Calc(b float64, x float64) float64 {
	return (1 + math.Pow(math.Sin(math.Pow(b, 3)+math.Pow(x, 3)), 2)) / (math.Cbrt(math.Pow(b, 3) + math.Pow(x, 3)))
}

func TaskA(b float64, Xn float64, Xk float64, delX float64) {
	for x := Xn; x <= Xk; x += delX {
		fmt.Println("При x =", x, "y =", Calc(b, x))
	}
}

func TaskB(b float64, x [5]float64) {
	for _, value := range x {
		fmt.Println("При x =", value, "y =", Calc(b, value))
	}
}

File: python/lab4.py
import math


def CalculateFunction(x, b):
    y = (1 + (math.sin(b**3 + x**3))**2) / ((b**3 + x**3)**(1/3))
    return y


def TaskA(xa_beginning, xa_end, xa_delta, b):
    y_a = []
    while xa_beginning != (xa_end + xa_delta):
        y_a.append(CalculateFunction(xa_beginning, b))
        xa_beginning += xa_delta
    return y_a


def TaskB(xs, b):
    y_b = []
    for i in xs:
        y_b.append(CalculateFunction(i, b))
    return y_b


print("Задача А", TaskA(1.28, 3.28, 0.4, 2.5))
xs = [1.1, 2.4, 3.6, 1.7, 3.9]
print("Задача В", TaskB(xs, 2.5))
